src/main/java/rx/internal/operators/OperatorMerge.java

```diff
@@ -213,7 +213,7 @@ private void handleNewSource(Observable<? extends T> t) {
             InnerSubscriber<T> i = new InnerSubscriber<T>(this, producerIfNeeded);
             i.sindex = childrenSubscribers.add(i);
             t.unsafeSubscribe(i);
-            if (!isUnsubscribed()) {
+            if ((producerIfNeeded == null || producerIfNeeded.requested > 0) && !isUnsubscribed()) {
```

Comment:
Merge deliberately doesn't limit the source observable of observables; this is what ```MergeMaxConcurrent``` is for. Besides, I can't rule out a race for ```requested```.